testutil/bq.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil contains common util functions to facilitate tests.
+package testutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/api/iterator"
+)
+
+// QueryBQEntries queries the BQ and checks if a bigqury entry matched the query exists or not and return the results.
+// [T any] needs to be bigquery iteration supported type (e.g. string, bigquery.Value, struct{}).
+func QueryBQEntries[T any](ctx context.Context, query *bigquery.Query) ([]*T, error) {
```

Comment:
So my understanding was [that interface types were always pointers,](https://jordanorelli.com/post/32665860244/how-to-use-interfaces-in-go) but seeing []*T being returned here seems to make me think that maybe that is not the case. 

Is it that on compilation time, any call to QueryBQEntries will have T be wrapped/unwrapped in an interface, so the []*T is still needed, as outside of the context of QueryBQEntries, T is still a concerte type?